code/client/src/components/LocalImport.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react'
+import { ImportOutlined, LoadingOutlined } from '@ant-design/icons'
+import { Button, Upload } from 'antd'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router'
+import message from '../message'
+import storage from '../storage'
+import util from '../util'
+import ONEUtil from '../../../lib/util'
+import { walletActions } from '../state/modules/wallet'
+import Paths from '../constants/paths'
+
+
+const LocalImport = ({onSuccess}) => {
+  const dispatch = useDispatch()
+  const history = useHistory()
+  const wallets = useSelector(state => state.wallet.wallets)
+  const [fileUploading, setFileUploading] = useState(false)
+
+  const getDataFromFile = file =>
+    new Promise(resolve => {
+      const reader = new FileReader()
+      reader.addEventListener('load', () => resolve(reader.result.split('\r\n')))
+      reader.readAsText(file)
+    })
+
+  const handleImport = async info => {
+    if (info.file.status === 'uploading') {
+      setFileUploading(true)
+    }
+
+    if (info.file.status === 'done') {
+      try {
+        const [walletString, ...buffer] = await getDataFromFile(info.file.originFileObj)
+        const wallet = JSON.parse(walletString)
+        const layers = buffer.map(b => new Uint8Array(b.split(',')))
```

Comment:
that would be very inefficient - I suggest using protobuf or something here

Reply:
See https://github.com/protobufjs/protobuf.js/ for encoding and intro

See https://github.com/polymorpher/one-wallet/blob/master/code/client/src/proto/oauth_migration.proto for proto example

See https://github.com/polymorpher/one-wallet/blob/master/code/client/package.json#L10 for compiling

See https://github.com/polymorpher/one-wallet/blob/master/code/client/src/components/OtpTools.jsx#L31 for decoding

Reply:
Yep, the current exported files are too huge too, protobuf should able to reduce the size and faster when parsing it. I just upload a WIP version because I need more time on the better solution.

Reply:
I also wrote a serializer in CLI version without using protobuf: https://github.com/polymorpher/one-wallet/blob/master/code/cli/src/store.js#L41 - that works too but you would need to encode the state part as binary also